### PR TITLE
Add Kos solution

### PR DESCRIPTION
**Pull request by cdragan**
## Description
This solution is written in Kos, which is a scripting language.

It uses a class to encapsulate the sieve, using a `buffer` type, which is an array of bytes.

Flags for subsequent primes are stored in bytes (8 bits), which is much faster than doing bit manipulation in an interpreted language, without using any module specific to handling sequences of bits.

This is a single-threaded solution (a multi-threaded solution is possible, but I haven't explored it yet).

The solutions uses a base faithful algorithm, including the allowed optimizations.

## Contributing requirements
* [x] I read the contribution guidelines in CONTRIBUTING.md.
* [x] I placed my solution in the correct solution folder.
* [x] I added a README.md with the right badge(s).
* [x] I added a Dockerfile that builds and runs my solution.
* [x] I selected `drag-race` as the target branch.
* [x] All code herein is licensed compatible with BSD-3.

### CODE REVIEW

File: PrimeKos/solution_1/primes.kos [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                }
+            }
+        }
+
+        # Count remaining primes
+        for const factor in range(max_clear_factor + 1, sieve.size) {
+            if ! sieve[factor] {
+                num_primes += 1
+            }
+        }
```
Comment (rbergen) [inline PrimeKos/solution_1/primes.kos]: It's not necessary to keep track of/count the number of primes found when running the sieve. In fact, doing so could be considered a waste of clock cycles during the timed run. Instead, in a faithful implementation, the function that runs the sieve should not return the number of primes found. but the sieve array itself.

**Review by rbergen — APPROVED** [review]

File: PrimeKos/solution_1/primes.kos (hunk 1 above)
Comment (cdragan) [reply PrimeKos/solution_1/primes.kos]: Thanks a lot, fantastic, let me fix that. :-)

Comment (cdragan) [reply PrimeKos/solution_1/primes.kos]: Updated.

File: PrimeKos/solution_1/primes.kos [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+        # Look from primes starting with 3 up to square root of sieve size
+        # Note we divide by 2 since we keep only odd numbers
+        for const factor in range(1, max_clear_factor + 1) {
+
+            # If this slot has 0, it means it's not a multiple of a prime,
+            # therefore it corresponds to a prime number
+            if ! sieve[factor] {
+
+                num_primes += 1
```
Comment (rbergen) [inline PrimeKos/solution_1/primes.kos]: Is this still necessary?

File: PrimeKos/solution_1/primes.kos [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+                const prime = factor * 2 + 1
+
+                # Mark multiples of this prime, starting at prime squared
+                for const clear_factor in range(prime * prime / 2, sieve.size, prime) {
+                    sieve[clear_factor] = 1
+                }
+            }
+        }
+    }
```
Comment (rbergen) [inline PrimeKos/solution_1/primes.kos:44]: It seems the run function is not returning the sieve array/buffer, yet.

File: PrimeKos/solution_1/primes.kos (hunk 2 above)
Comment (cdragan) [reply PrimeKos/solution_1/primes.kos]: Thanks, it is not. I've just noticed this as well. Fix in flight.

File: PrimeKos/solution_1/primes.kos (hunk 3 above)
Comment (cdragan) [reply PrimeKos/solution_1/primes.kos:44]: The sieve buffer/array is the member of the class. I don't think we need to return it from the run function.

Comment (cdragan) [reply PrimeKos/solution_1/primes.kos:44]: Updated to return the sieve buffer/array from the run function.

**Review by rbergen — APPROVED** [review]
LGTM

**Review by marghidanu — APPROVED** [review]

**Pull request merged**